### ISSUE TITLE
Interactive save should respect 'savefig.facecolor' rcParam.

### ISSUE THREAD

**Issue by u55** [matplotlib/matplotlib#3437]
Hi matplotlib developers,

Currently, there is no way to set the figure facecolor when saving a figure interactively via the `NavigationToolbar2` graphical save button.  All interactive backends, that I have tested, default to a white facecolor while ignoring the 'savefig.facecolor' rcParam.  I propose to change this behavior such that every backend will respect the 'savefig.facecolor' rcParam.  

One possible option would be to edit the `save_figure` method of each backend-specific class which inherits from `matplotlib.backend_bases.NavigationToolbar2` in every backend.  This seems tedious.

Another option would be to edit the `print_figure` method in `matplotlib.backend_bases.FigureCanvasBase`, which every backend inherits from.  This seems simpler.

For instance, adding the following indicated lines to `matplotlib.backend_bases.FigureCanvasBase.print_figure` works for me.

```python
def print_figure(self, filename, dpi=None, facecolor=None, edgecolor='w', # <---- facecolor default changed to None
                     orientation='portrait', format=None, **kwargs):

    ...

    if dpi is None:
        dpi = rcParams['savefig.dpi']

    if facecolor is None:  #  <---- added
        facecolor = rcParams['savefig.facecolor']  # <---- added

    origDPI = self.figure.dpi
    origfacecolor = self.figure.get_facecolor()
    origedgecolor = self.figure.get_edgecolor()

    self.figure.dpi = dpi
    self.figure.set_facecolor(facecolor)
    self.figure.set_edgecolor(edgecolor)

    ...
```

I don't use a custom 'edgecolor' often, but the interactive figure saving functionality should also be changed to use the 'savefig.edgecolor' rcParam.

I would be happy to create a pull request if the developers agree that this change is a good idea.
Thank you.

**Comment by braxtonj** [comment]
Has anyone found a workaround for this in the meantime?  No matter what I try, my interactive plots are saving with a white background which makes all of the labels completely unreadable.  Do I have to waste time redesigning my color scheme or can someone point me in the right direction.

This is what I have thusfar:

```python
lib_mpl.rcParams['savefig.dpi'] = graphDPI
lib_mpl.rcParams['savefig.facecolor'] = figbgColor
lib_mpl.rcParams['savefig.edgecolor'] = figbgColor
f,ax = lib_plt.subplots(facecolor=figbgColor)
```

The dpi works just find but not the savefig.facecolor nor savefig.edgecolor

Thanks,
Braxton

**Comment by u55** [comment]
@braxtonj,  I believe that the only two workarounds are: (1) save the figure programmatically via the `savefig` method rather than saving interactively, or (2) modify the source code to the file `matplotlib.backend_bases.FigureCanvasBase.print_figure` as suggested in my first post.

**Comment by tacaswell** [comment]
@u55 Can you put in a PR with that change?

**Comment by mgrady3** [comment]
Has any further work been done on this PR?

I know there's a major update to mpl in the works (heard about it in various pycon/scipy talks).
Will this issue be addressed in the next edition of mpl?

For GUI programs that are already using the built in mpl toolbar it would be nice to have the savefig button on the toolbar respect rcParams rather than having to manually add another GUI element to place a call programmatically to savefig() for the sole purpose of changing the facecolor of the saved image and thus essentially having a dead item in the built in toolbar ...

**Comment by tacaswell** [comment]
ping @u55

**Comment by tacaswell** [comment]
closed by #6197
